05/thbwd/dead_code2.oat

```diff
@@ -0,0 +1,6 @@
+int program(int argc, string[] argv) {
+  for (var i = 0; i < 40; return 3;) {
+    print_string("I'm a zombie");
```

Comment:
I'd argue that the type error must be thrown because there's a statement in the 'stmt_opt' part which definitely returns. The rule 'TYP_FOR' requires that this stmt might not return.

Reply:
Yes, that's exactly what I'm testing. This file is in `typecheck_file_error`.

Reply:
I was more thinking about not calling this dead code. If, for example, the block was empty we'd not have dead code, but this would still be invalid because of the invalid premise for the **TYP_FOR** rule. I'd be okay with removing line 3 as then it's the minimum that's necessary for this violation to occur :)

Reply:
I see what you mean. Indeed, this would even print "I'm a zombie" once if allowed by the type checker. I can rename the file when I find time.

Reply:
Thank you!